tensorflow/contrib/seq2seq/python/ops/beam_search_decoder.py

```diff
@@ -622,6 +629,10 @@ def _beam_search_step(time, logits, next_cell_state, beam_state, batch_size,
       range_size=beam_width,
       gather_shape=[-1])
   next_prediction_len += lengths_to_add
+  next_prediction_len = array_ops.where(next_beam_probs > -np.Inf,
```

Comment:
can this happen anywhere except t=0?

Reply:
Do you mean if `next_beam_probs > -np.Inf` can be evaluated to be True if t>0? it can.

Or maybe you mean `next_beam_probs > -np.Inf` can only be evaluated to be False when t=0? 
well, when vocabulary is small then it could be evaluated to be False. I was trying to use your Beamsearch to the test phase of pointer network, and the vocabulary can't be determined when building the graph. and it usually is small.

Reply:
I think this can be true when t>0. I think it is generally ok since as long as vocab_size != 1, the available beam width will expand towards to the beam_width.

I think we can be more careful about `finished`, for example, if all finished == False entries' log score is -neg_inf, we terminate the decoding process. This can be done in the `step()` method.  wdyt?

Reply:
@JerrikEph thoughts?

Reply:
This should be handled by init self._finished as initial log_probs.